packages/migrate/src/MigrateEngine.ts

```diff
@@ -135,11 +142,29 @@ export class MigrateEngine {
     return this.runCommand(this.getRPCPayload('evaluateDataLoss', args))
   }
 
+  public getDatabaseDescription(schema: string): Promise<string> {
+    return this.runCommand(this.getRPCPayload('getDatabaseDescription', { schema }))
+  }
+
   /**
    * Get the database version for error reporting.
    */
-  public getDatabaseVersion(): Promise<string> {
-    return this.runCommand(this.getRPCPayload('getDatabaseVersion', undefined))
+  public getDatabaseVersion({ schema }: EngineArgs.GetDatabaseVersionParams): Promise<string> {
+    return this.runCommand(this.getRPCPayload('getDatabaseVersion', { schema }))
+  }
+
+  /**
+   * Given a Prisma schema, introspect the database definitions and update the schema with the results.
+   * `compositeTypeDepth` is optional, and only required for MongoDB.
+   * TODO: change `compositeTypeDepth` to -1 once https://github.com/prisma/prisma-engines/pull/3487 is merged.
```

Comment:
PR https://github.com/prisma/prisma-engines/pull/3487 got dismissed

Reply:
I had done that and reverted it in https://github.com/prisma/prisma/pull/16766#:~:text=%E2%80%A6-,392a6d2,-jkomyno%20added%20this, as apparently that PR alone doesn't fix this type mismatch problem in JSON-RPC between JS and Rust